kernel: bump 5.10 to 5.10.96

rsalvaterra opened the pull request:
Deleted (upstreamed):
bcm27xx/patches-5.10/950-0669-drm-vc4-hdmi-Make-sure-the-device-is-powered-with-CE.patch [1]
bcm27xx/patches-5.10/950-0672-drm-vc4-hdmi-Move-initial-register-read-after-pm_run.patch [1]
gemini/patches-5.10/0003-ARM-dts-gemini-NAS4220-B-fis-index-block-with-128-Ki.patch [2]

Manually rebased:
bcm27xx/patches-5.10/950-0675-drm-vc4-hdmi-Drop-devm-interrupt-handler-for-CEC-int.patch

[1] https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v5.10.94&id=55b10b88ac8654fc2f31518aa349a2e643b37f18
[2] https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v5.10.94&id=958a8819d41420d7a74ed922a09cacc0ba3a4218

Run-tested:
ramips/mt7621 (Redmi AC2100)

Signed-off-by: Rui Salvaterra \<rsalvaterra@gmail.com\>

rsalvaterra committed:
kernel: bump 5.10 to 5.10.93

No patches required rebasing.

Signed-off-by: Rui Salvaterra <rsalvaterra@gmail.com>

ldir-EDB0 commented:
My APU2 hasn't exploded, so that's X86_64 ok

graysky2 commented:
```
Build system: x86_64
Build-tested: bcm2711/RPi4B
Run-tested: bcm2711/RPi4B
```

rsalvaterra committed:
kernel: bump 5.10 to 5.10.94

Deleted (upstreamed):
bcm27xx/patches-5.10/950-0669-drm-vc4-hdmi-Make-sure-the-device-is-powered-with-CE.patch [1]
bcm27xx/patches-5.10/950-0672-drm-vc4-hdmi-Move-initial-register-read-after-pm_run.patch [1]
gemini/patches-5.10/0003-ARM-dts-gemini-NAS4220-B-fis-index-block-with-128-Ki.patch [2]

Manually rebased:
bcm27xx/patches-5.10/950-0675-drm-vc4-hdmi-Drop-devm-interrupt-handler-for-CEC-int.patch

Manually reverted:
generic/pending-5.10/860-Revert-ASoC-mediatek-Check-for-error-clk-pointer.patch [3]

[1] https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v5.10.94&id=55b10b88ac8654fc2f31518aa349a2e643b37f18
[2] https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v5.10.94&id=958a8819d41420d7a74ed922a09cacc0ba3a4218
[3] https://lore.kernel.org/all/trinity-2a727d96-0335-4d03-8f30-e22a0e10112d-1643363480085@3c-app-gmx-bap33/

Signed-off-by: Rui Salvaterra <rsalvaterra@gmail.com>
Signed-off-by: Daniel Golle <daniel@makrotopia.org>

graysky2 commented:
```
Build system: x86_64
Build-tested: bcm2711/RPi4B
Run-tested: bcm2711/RPi4B
```

@rsalvaterra  - I also built and flashed flashed an image for `mt7622/RT3200` however, it rendered the RT3200 in a state without a functional network (hardwire or wireless).

* I could not ping it.
* I use the WAN port as the uplink to my router and that LED was not illuminated so I assume the link was down. 
 
I put it into [failsafe mode](https://forum.openwrt.org/t/belkin-rt3200-linksys-e8450-wifi-ax-discussion/94302/1354), but again could not ping it.  I recovered by holding reset+power on, flashing my last-known-good image (5.10.93 from a week ago), then restored by backup settings and it is working again.

I do not know if the breakage occurred as a function of this PR or of other changes made to master since I last built 5.10.93.  I built that image on 20-Jan-2022 but if memory serves me correctly, I wanted to avoid the firewall 4 change in https://github.com/openwrt/openwrt/commit/08d9f6e3020b4a149b2007b6ed7d684c49af9bbf so I *think* built it from a commit or two BEFORE that one.

rsalvaterra commented:
> @rsalvaterra - I also built and flashed flashed an image for `mt7622/RT3200` however, it rendered the RT3200 in a state without a functional network (hardwire or wireless).

Yikes, that's not good. 😱 However, @dangowrt has been doing quite a lot of work on that exact machine, maybe he has some idea of what might be happening…? 😕

graysky2 commented:
> have you tried using the serial console to find the smoking gun?

No, I did not open the RT3200 to try a serial port, nor do I have the hardware needed to connect to it.  I just did the recovery process blinded and fortunately for me, it came back up on the older image.  I am reluctant to try more experimentation with it for fear of bricking the thing.

graysky2 commented:
OK... I built from the latest master without reverting https://github.com/openwrt/openwrt/commit/08d9f6e3020b4a149b2007b6ed7d684c49af9bbf and applying your https://github.com/openwrt/openwrt/pull/4975/commits/a77fa4ca3287e795d88b6bfd8b3d211fa38299a2 (since I know that 5.10.93 worked in the past).  I booted fine and connectivity is fine.

graysky2 commented:
@dangowrt - I do not have a serial connector in order to connect.

graysky2 commented:
OK... the problem is definitely [kernel: bump 5.10 to 5.10.94](https://github.com/openwrt/openwrt/commit/0b39b9a231138d162727cd119d2ead149a09b4b7)

* The latest master without reverting https://github.com/openwrt/openwrt/commit/08d9f6e3020b4a149b2007b6ed7d684c49af9bbf and applying [kernel: bump 5.10 to 5.10.93](https://github.com/openwrt/openwrt/pull/4975/commits/a77fa4ca3287e795d88b6bfd8b3d211fa38299a2) worked fine.
* Applying [kernel: bump 5.10 to 5.10.94](https://github.com/openwrt/openwrt/commit/0b39b9a231138d162727cd119d2ead149a09b4b7) on top broke the RT3200.

I think it boots but no network or radios come up.  I see the power LED glow solid for a few seconds then begin to slow blink, then fast blink, then slow blink, then finally glow solid.  But again, the WAN light never lights up (uplink port for me) and the radios do not come up.  Had to hold reset + power on to recover.

@dangowrt - Does narrowing it down the the 5.10.94 kernel help you debug in any way?

graysky2 commented:
@rsalvaterra - Yes, very likely in 5.10.94 itself according to the above post by @dangowrt 

> There have been quite some changes from v5.10.93 to v5.10.94. Relevant for this device there are a bunch of MTD-related things, an innocent looking error-handling improvement for mtk_eth_soc and some mm fixes related to DMA

justinkb commented:
I just built master with this PR applied on top of it, here is the log from the serial port of my Belkin RT3200

```

F0: 102B 0000
F6: 0000 0000
V0: 0000 0000 [0001]
00: 0000 0000
BP: 0400 0041 [0000]
G0: 1190 0000
T0: 0000 02AE [000F]
Jump to BL

NOTICE:  BL2: v2.4(release):OpenWrt v2021-05-08-d2c75b21-1 (mt7622-snand-1ddr)
NOTICE:  BL2: Built : 22:51:20, Nov 19 2021
NOTICE:  SPI-NAND: FM35Q1GA (128MB)
NOTICE:  BL2: Booting BL31
NOTICE:  BL31: v2.4(release):OpenWrt v2021-05-08-d2c75b21-1 (mt7622-snand-1ddr)
NOTICE:  BL31: Built : 22:51:20, Nov 19 2021
serial_mtk serial@11002000: pinctrl_select_state_full: uclass_get_device_by_phandle_id: err=-19


U-Boot 2021.10 (Nov 19 2021 - 22:51:20 +0000)

CPU:   MediaTek MT7622
Model: mt7622-linksys-e8450-ubi
DRAM:  512 MiB
WDT:   Started with servicing (60s timeout)
MMC:
Loading Environment from UBI... SPI-NAND: FM35Q1GA (128MB)
ubi0 warning: print_rsvd_warning: cannot reserve enough PEBs for bad PEB handling, reserved 18, need 20
Read 126976 bytes from volume ubootenv to 000000005f7bf3c0
Read 126976 bytes from volume ubootenv2 to 000000005f7de400
OK
In:    serial@11002000
Out:   serial@11002000
Err:   serial@11002000
reset button found
Loading Environment from UBI... ubi0 warning: print_rsvd_warning: cannot reserve enough PEBs for bad PEB handling, reserved 18, need 20
Read 126976 bytes from volume ubootenv to 000000005f7bf3c0
Read 126976 bytes from volume ubootenv2 to 000000005f7de400
OK
Net:   eth0: ethernet@1b100000

      ( ( ( OpenWrt ) ) )       U-Boot 2021.10 (Nov 19 2021 - 22:51:20 +0000)

     1. Run default boot command.
     2. Boot system via TFTP.
     3. Boot production system from flash.
     4. Boot recovery system from flash.
     5. Load production system via TFTP then write to flash.
     6. Load recovery system via TFTP then write to flash.
     7. Load BL31+U-Boot FIP via TFTP then write to flash.
     8. Load BL2 preloader via TFTP then write to flash.
     9. Reboot.
     a. Reset all settings to factory defaults.
     0. U-Boot console


  Press UP/DOWN to move, ENTER to select, ESC/CTRL+C to quit
ubi0 warning: print_rsvd_warning: cannot reserve enough PEBs for bad PEB handling, reserved 18, need 20
No size specified -> Using max size (10031104)
Read 10031104 bytes from volume fit to 0000000048000000

## Checking Image at 48000000 ...
   FIT image found
   FIT description: ARM64 OpenWrt FIT (Flattened Image Tree)
    Image 0 (kernel-1)
     Description:  ARM64 OpenWrt Linux-5.10.94
     Type:         Kernel Image
     Compression:  gzip compressed
     Data Start:   0x48001000
     Data Size:    5310916 Bytes = 5.1 MiB
     Architecture: AArch64
     OS:           Linux
     Load Address: 0x44000000
     Entry Point:  0x44000000
     Hash algo:    crc32
     Hash value:   9afb3b92
     Hash algo:    sha1
     Hash value:   f1b8afcc2bbedcbcc031e66c21cd6f8600b59f2e
    Image 1 (fdt-1)
     Description:  ARM64 OpenWrt linksys_e8450-ubi device tree blob
     Type:         Flat Device Tree
     Compression:  uncompressed
     Data Start:   0x48512000
     Data Size:    29481 Bytes = 28.8 KiB
     Architecture: AArch64
     Hash algo:    crc32
     Hash value:   cbb7947d
     Hash algo:    sha1
     Hash value:   af69d945d955d91d686d5fa1c20ba22ef9e59d33
    Image 2 (rootfs-1)
     Description:  ARM64 OpenWrt linksys_e8450-ubi rootfs
     Type:         Filesystem Image
     Compression:  uncompressed
     Data Start:   0x4851a000
     Data Size:    4583424 Bytes = 4.4 MiB
     Hash algo:    crc32
     Hash value:   109eddac
     Hash algo:    sha1
     Hash value:   ff5fa99db1df153818042b91ca1be31d638d1a8f
    Default Configuration: 'config-1'
    Configuration 0 (config-1)
     Description:  OpenWrt linksys_e8450-ubi
     Kernel:       kernel-1
     FDT:          fdt-1
     Loadables:    rootfs-1
## Checking hash(es) for FIT Image at 48000000 ...
   Hash(es) for Image 0 (kernel-1): crc32+ sha1+
   Hash(es) for Image 1 (fdt-1): crc32+ sha1+
   Hash(es) for Image 2 (rootfs-1): crc32+ sha1+
## Loading kernel from FIT Image at 48000000 ...
   Using 'config-1' configuration
   Trying 'kernel-1' kernel subimage
     Description:  ARM64 OpenWrt Linux-5.10.94
     Type:         Kernel Image
     Compression:  gzip compressed
     Data Start:   0x48001000
     Data Size:    5310916 Bytes = 5.1 MiB
     Architecture: AArch64
     OS:           Linux
     Load Address: 0x44000000
     Entry Point:  0x44000000
     Hash algo:    crc32
     Hash value:   9afb3b92
     Hash algo:    sha1
     Hash value:   f1b8afcc2bbedcbcc031e66c21cd6f8600b59f2e
   Verifying Hash Integrity ... crc32+ sha1+ OK
## Loading fdt from FIT Image at 48000000 ...
   Using 'config-1' configuration
   Trying 'fdt-1' fdt subimage
     Description:  ARM64 OpenWrt linksys_e8450-ubi device tree blob
     Type:         Flat Device Tree
     Compression:  uncompressed
     Data Start:   0x48512000
     Data Size:    29481 Bytes = 28.8 KiB
     Architecture: AArch64
     Hash algo:    crc32
     Hash value:   cbb7947d
     Hash algo:    sha1
     Hash value:   af69d945d955d91d686d5fa1c20ba22ef9e59d33
   Verifying Hash Integrity ... crc32+ sha1+ OK
   Booting using the fdt blob at 0x48512000
## Loading loadables from FIT Image at 48000000 ...
   Trying 'rootfs-1' loadables subimage
     Description:  ARM64 OpenWrt linksys_e8450-ubi rootfs
     Type:         Filesystem Image
     Compression:  uncompressed
     Data Start:   0x4851a000
     Data Size:    4583424 Bytes = 4.4 MiB
     Hash algo:    crc32
     Hash value:   109eddac
     Hash algo:    sha1
     Hash value:   ff5fa99db1df153818042b91ca1be31d638d1a8f
   Verifying Hash Integrity ... crc32+ sha1+ OK
   Uncompressing Kernel Image
   Loading Device Tree to 000000005f7f1000, end 000000005f7fb328 ... OK
Add 'reserved-memory' node failed: FDT_ERR_EXISTS

Starting kernel ...

[    0.000000] Booting Linux on physical CPU 0x0000000000 [0x410fd034]
[    0.000000] Linux version 5.10.94 (builder@buildhost) (aarch64-openwrt-linux-musl-gcc (OpenWrt GCC 11.2.0 r18642+7-ef4c97576b) 11.2.0, GNU ld (GNU Binutils) 2.37) #0 SMP Fri Jan 28 09:43:32 2022
[    0.000000] Machine model: Linksys E8450 (UBI)
[    0.000000] earlycon: uart8250 at MMIO32 0x0000000011002000 (options '')
[    0.000000] printk: bootconsole [uart8250] enabled
[    0.000000] Zone ranges:
[    0.000000]   DMA      [mem 0x0000000040000000-0x000000005fffffff]
[    0.000000]   DMA32    empty
[    0.000000]   Normal   empty
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000040000000-0x0000000042ffffff]
[    0.000000]   node   0: [mem 0x0000000043000000-0x000000004302ffff]
[    0.000000]   node   0: [mem 0x0000000043030000-0x000000005fffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000040000000-0x000000005fffffff]
[    0.000000] On node 0 totalpages: 131072
[    0.000000]   DMA zone: 2048 pages used for memmap
[    0.000000]   DMA zone: 0 pages reserved
[    0.000000]   DMA zone: 131072 pages, LIFO batch:31
[    0.000000] psci: probing for conduit method from DT.
[    0.000000] psci: PSCIv1.1 detected in firmware.
[    0.000000] psci: Using standard PSCI v0.2 function IDs
[    0.000000] psci: MIGRATE_INFO_TYPE not supported.
[    0.000000] psci: SMC Calling Convention v1.2
[    0.000000] percpu: Embedded 20 pages/cpu s43864 r8192 d29864 u81920
[    0.000000] pcpu-alloc: s43864 r8192 d29864 u81920 alloc=20*4096
[    0.000000] pcpu-alloc: [0] 0 [0] 1
[    0.000000] Detected VIPT I-cache on CPU0
[    0.000000] CPU features: detected: ARM erratum 845719
[    0.000000] CPU features: kernel page table isolation disabled by kernel configuration
[    0.000000] CPU features: detected: ARM erratum 843419
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 129024
[    0.000000] Kernel command line: earlycon=uart8250,mmio32,0x11002000 console=ttyS0,115200n1 swiotlb=512
[    0.000000] Dentry cache hash table entries: 65536 (order: 7, 524288 bytes, linear)
[    0.000000] Inode-cache hash table entries: 32768 (order: 6, 262144 bytes, linear)
[    0.000000] mem auto-init: stack:off, heap alloc:off, heap free:off
[    0.000000] Memory: 493424K/524288K available (8126K kernel code, 882K rwdata, 2228K rodata, 448K init, 300K bss, 30864K reserved, 0K cma-reserved)
[    0.000000] SLUB: HWalign=64, Order=0-3, MinObjects=0, CPUs=2, Nodes=1
[    0.000000] rcu: Hierarchical RCU implementation.
[    0.000000]  Tracing variant of Tasks RCU enabled.
[    0.000000] rcu: RCU calculated value of scheduler-enlistment delay is 10 jiffies.
[    0.000000] NR_IRQS: 64, nr_irqs: 64, preallocated irqs: 0
[    0.000000] GIC: GICv2 detected, but range too small and irqchip.gicv2_force_probe not set
[    0.000000] random: get_random_bytes called from start_kernel+0x340/0x484 with crng_init=0
[    0.000000] arch_timer: cp15 timer(s) running at 12.50MHz (phys).
[    0.000000] clocksource: arch_sys_counter: mask: 0xffffffffffffff max_cycles: 0x2e2049cda, max_idle_ns: 440795202628 ns
[    0.000003] sched_clock: 56 bits at 12MHz, resolution 80ns, wraps every 4398046511080ns
[    0.008492] Calibrating delay loop (skipped), value calculated using timer frequency.. 25.00 BogoMIPS (lpj=125000)
[    0.019223] pid_max: default: 32768 minimum: 301
[    0.024080] Mount-cache hash table entries: 1024 (order: 1, 8192 bytes, linear)
[    0.031658] Mountpoint-cache hash table entries: 1024 (order: 1, 8192 bytes, linear)
[    0.040657] rcu: Hierarchical SRCU implementation.
[    0.045749] dyndbg: Ignore empty _ddebug table in a CONFIG_DYNAMIC_DEBUG_CORE build
[    0.053970] smp: Bringing up secondary CPUs ...
[    0.058985] Detected VIPT I-cache on CPU1
[    0.058992] CPU features: SANITY CHECK: Unexpected variation in SYS_CNTFRQ_EL0. Boot CPU: 0x00000000bebc20, CPU1: 0x00000000000000
[    0.059007] CPU features: Unsupported CPU feature variation detected.
[    0.059040] CPU1: Booted secondary processor 0x0000000001 [0x410fd034]
[    0.059104] smp: Brought up 1 node, 2 CPUs
[    0.093088] SMP: Total of 2 processors activated.
[    0.097956] CPU features: detected: 32-bit EL0 Support
[    0.103278] CPU features: detected: CRC32 instructions
[    0.108696] CPU: All CPU(s) started at EL2
[    0.112940] alternatives: patching kernel code
[    0.121071] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.131290] futex hash table entries: 512 (order: 3, 32768 bytes, linear)
[    0.138405] pinctrl core: initialized pinctrl subsystem
[    0.144922] NET: Registered protocol family 16
[    0.149950] DMA: preallocated 128 KiB GFP_KERNEL pool for atomic allocations
[    0.157304] DMA: preallocated 128 KiB GFP_KERNEL|GFP_DMA pool for atomic allocations
[    0.165341] DMA: preallocated 128 KiB GFP_KERNEL|GFP_DMA32 pool for atomic allocations
[    0.173907] thermal_sys: Registered thermal governor 'fair_share'
[    0.173914] thermal_sys: Registered thermal governor 'bang_bang'
[    0.180227] thermal_sys: Registered thermal governor 'step_wise'
[    0.186479] thermal_sys: Registered thermal governor 'user_space'
[    0.192985] ASID allocator initialised with 65536 entries
[    0.205425] pstore: Registered ramoops as persistent store backend
[    0.211859] ramoops: using 0x10000@0x42ff0000, ecc: 0
[    0.248853] SCSI subsystem initialized
[    0.253035] libata version 3.00 loaded.
[    0.257168] usbcore: registered new interface driver usbfs
[    0.262968] usbcore: registered new interface driver hub
[    0.268515] usbcore: registered new device driver usb
[    0.275954] clocksource: Switched to clocksource arch_sys_counter
[    0.282945] NET: Registered protocol family 2
[    0.287613] IP idents hash table entries: 8192 (order: 4, 65536 bytes, linear)
[    0.295557] tcp_listen_portaddr_hash hash table entries: 256 (order: 0, 4096 bytes, linear)
[    0.304250] TCP established hash table entries: 4096 (order: 3, 32768 bytes, linear)
[    0.312318] TCP bind hash table entries: 4096 (order: 4, 65536 bytes, linear)
[    0.319758] TCP: Hash tables configured (established 4096 bind 4096)
[    0.326424] UDP hash table entries: 256 (order: 1, 8192 bytes, linear)
[    0.333195] UDP-Lite hash table entries: 256 (order: 1, 8192 bytes, linear)
[    0.340514] NET: Registered protocol family 1
[    0.345039] PCI: CLS 0 bytes, default 64
[    0.350285] workingset: timestamp_bits=46 max_order=17 bucket_order=0
[    0.359842] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    0.365889] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
[    0.408999] Block layer SCSI generic (bsg) driver version 0.4 loaded (major 250)
[    0.417215] mtk-tphy 1a0c4000.usb-phy: failed to get ref_clk(id-1)
[    0.429098] mt-pmic-pwrap 10001000.pwrap: unexpected interrupt int=0x1
[    0.430102] mtk-scpsys: probe of 10006000.power-controller failed with error -2
[    0.445248] Serial: 8250/16550 driver, 16 ports, IRQ sharing enabled
[    0.453996] printk: console [ttyS0] disabled
[    0.458539] 11002000.serial: ttyS0 at MMIO 0x11002000 (irq = 22, base_baud = 1562500) is a 16550A
[    0.467763] printk: console [ttyS0] enabled
[    0.467763] printk: console [ttyS0] enabled
[    0.476131] printk: bootconsole [uart8250] disabled
[    0.476131] printk: bootconsole [uart8250] disabled
[    0.486550] 11004000.serial: ttyS1 at MMIO 0x11004000 (irq = 23, base_baud = 1562500) is a 16550A
[    0.496328] 1100c000.serial: ttyS2 at MMIO 0x1100c000 (irq = 27, base_baud = 17499995) is a MediaTek BTIF
[    0.506064] serial serial0: tty port ttyS2 registered
[    0.511804] mtk_rng 1020f000.rng: registered RNG driver
[    0.511816] hwrng: no data available
[    0.520770] cacheinfo: Unable to detect cache hierarchy for CPU 0
[    0.529990] loop: module loaded
[    0.533135] Loading iSCSI transport class v2.0-870.
[    0.539763] mtk-snand 1100d000.snfi: chip is FM35Q1GA, size 128MB, page size 2048, oob size 64
[    0.548628] 4 fixed-partitions partitions found on MTD device 1100d000.snfi
[    0.555588] Creating 4 MTD partitions on "1100d000.snfi":
[    0.561026] 0x000000000000-0x000000080000 : "bl2"
[    0.566664] 0x000000080000-0x0000001c0000 : "fip"
[    0.573001] 0x0000001c0000-0x0000002c0000 : "factory"
[    0.579477] 0x000000300000-0x000008000000 : "ubi"
[    0.613003] random: fast init done
[    0.721021] i2c /dev entries driver
[    0.725837] mtk-wdt 10212000.watchdog: Watchdog enabled (timeout=31 sec, nowayout=0)
[    0.737307] NET: Registered protocol family 10
[    0.742565] Segment Routing with IPv6
[    0.746303] NET: Registered protocol family 17
[    0.750772] bridge: filtering via arp/ip/ip6tables is no longer available by default. Update your scripts to load br_netfilter if you need this.
[    0.763875] 8021q: 802.1Q VLAN Support v1.8
[    0.769253] pstore: Using crash dump compression: deflate
[    0.785209] mtk-pcie 1a143000.pcie: deferred probe timeout, ignoring dependency
[    0.792549] mtk-pcie: probe of 1a143000.pcie failed with error -110
[    0.798958] mtk-pcie 1a145000.pcie: deferred probe timeout, ignoring dependency
[    0.806271] mtk-pcie: probe of 1a145000.pcie failed with error -110
[    0.812615] mtk_hsdma 1b007000.dma-controller: deferred probe timeout, ignoring dependency
[    0.820877] mtk_hsdma: probe of 1b007000.dma-controller failed with error -110
[    0.828261] mtk_soc_eth 1b100000.ethernet: deferred probe timeout, ignoring dependency
[    0.836182] mtk_soc_eth: probe of 1b100000.ethernet failed with error -110
[    0.843146] xhci-mtk 1a0c0000.usb: deferred probe timeout, ignoring dependency
[    0.850366] xhci-mtk: probe of 1a0c0000.usb failed with error -110
[    0.857010] UBI: auto-attach mtd3
[    0.860333] ubi0: attaching mtd3
[    1.247985] ubi0: scanning is finished
[    1.256107] ubi0: attached mtd3 (name "ubi", size 125 MiB)
[    1.261594] ubi0: PEB size: 131072 bytes (128 KiB), LEB size: 126976 bytes
[    1.268467] ubi0: min./max. I/O unit sizes: 2048/2048, sub-page size 2048
[    1.275246] ubi0: VID header offset: 2048 (aligned 2048), data offset: 4096
[    1.282216] ubi0: good PEBs: 1000, bad PEBs: 0, corrupted PEBs: 0
[    1.288305] ubi0: user volume: 6, internal volumes: 1, max. volumes count: 128
[    1.295517] ubi0: max/mean erase counter: 1/0, WL threshold: 4096, image sequence number: 1959998407
[    1.304641] ubi0: available PEBs: 0, total reserved PEBs: 1000, PEBs reserved for bad PEB handling: 20
[    1.314004] ubi0: background thread "ubi_bgt0d" started, PID 730
[    1.320916] FIT: Default configuration: "config-1" (OpenWrt linksys_e8450-ubi)
[    1.328177] FIT:           kernel sub-image 0x00001000..0x005119c3 "kernel-1" (ARM64 OpenWrt Linux-5.10.94)
[    1.338010] FIT:          flat_dt sub-image 0x00512000..0x00519328 "fdt-1" (ARM64 OpenWrt linksys_e8450-ubi device tree blob)
[    1.349406] FIT:       filesystem sub-image 0x0051a000..0x00978fff "rootfs-1" (ARM64 OpenWrt linksys_e8450-ubi rootfs)
[    1.360182] FIT: selecting configured loadable "rootfs-1" to be root filesystem
[    1.367486]  ubiblock0_4: p1(rootfs-1)
[    1.367805] block ubiblock0_4: created from ubi0:4(fit)
[    1.380227] VFS: Mounted root (squashfs filesystem) readonly on device 259:1.
[    1.387566] Freeing unused kernel memory: 448K
[    1.425991] Run /sbin/init as init process
[    1.430081]   with arguments:
[    1.433038]     /sbin/init
[    1.435734]   with environment:
[    1.438877]     HOME=/
[    1.441227]     TERM=linux
[    1.616432] init: Console is alive
[    1.619981] init: - watchdog -
[    1.901915] kmodloader: loading kernel modules from /etc/modules-boot.d/*
[    1.952644] kmodloader: done loading kernel modules from /etc/modules-boot.d/*
[    1.966334] init: - preinit -
ip: SIOCGIFFLAGS: No such device
[    2.197098] random: procd: uninitialized urandom read (4 bytes read)
Press the [f] key and hit [enter] to enter failsafe mode
Press the [1], [2], [3] or [4] key and hit [enter] to select the debug level
[    6.268199] UBIFS (ubi0:5): Mounting in unauthenticated mode
[    6.273987] UBIFS (ubi0:5): background thread "ubifs_bgt0_5" started, PID 805
[    6.297309] UBIFS (ubi0:5): recovery needed
[    6.356476] UBIFS (ubi0:5): recovery completed
[    6.360998] UBIFS (ubi0:5): UBIFS: mounted UBI device 0, volume 5, name "rootfs_data"
[    6.368846] UBIFS (ubi0:5): LEB size: 126976 bytes (124 KiB), min./max. I/O unit sizes: 2048 bytes/2048 bytes
[    6.378761] UBIFS (ubi0:5): FS size: 92565504 bytes (88 MiB, 729 LEBs), journal size 4698112 bytes (4 MiB, 37 LEBs)
[    6.389190] UBIFS (ubi0:5): reserved for root: 4372097 bytes (4269 KiB)
[    6.395798] UBIFS (ubi0:5): media format: w5/r0 (latest is w5/r0), UUID 93A0FAB7-989F-4EAF-9822-DB2F03230F33, small LPT model
[    6.408999] mount_root: switching to ubifs overlay
[    6.416411] overlayfs: "xino" feature enabled using 2 upper inode bits.
[    6.426265] urandom-seed: Seeding with /etc/urandom.seed
[    6.484797] procd: - early -
[    6.487813] procd: - watchdog -
[    7.034820] procd: - watchdog -
[    7.038976] procd: - ubus -
[    7.054646] random: ubusd: uninitialized urandom read (4 bytes read)
[    7.092720] random: ubusd: uninitialized urandom read (4 bytes read)
[    7.099318] random: ubusd: uninitialized urandom read (4 bytes read)
[    7.108926] procd: - init -
Please press Enter to activate this console.
[    7.367529] urngd: v1.0.2 started.
[    7.395255] random: crng init done
[    7.398728] random: 6 urandom warning(s) missed due to ratelimiting
[    7.423156] kmodloader: loading kernel modules from /etc/modules.d/*
[    7.472075] Loading modules backported from Linux version v5.15.8-0-g43e577d7a2cb
[    7.479593] Backport generated by backports.git v5.15.8-1-0-g83f664bb
[    7.540060] mt7622-wmac 18000000.wmac: deferred probe timeout, ignoring dependency
[    7.547699] mt7622-wmac: probe of 18000000.wmac failed with error -110
[    7.576812] PPP generic driver version 2.4.2
[    7.581896] NET: Registered protocol family 24
[    7.590513] kmodloader: done loading kernel modules from /etc/modules.d/*
```

after this, nothing else comes over the port

graysky2 commented:
@justinkb - Thanks for posting that... I assume your device boots to a solid LED but no network/wireless as well.

> note that this is the first time I've built my own openwrt image, there might be something I did wrong in building it, in which case this wouldn't be relevant for debugging graysky2's problem

As a control experiment, just revert this PR and rebuild the image (so you will have the old kernel version 5.10.92 with the latest master).  If it boots and runs successfully, that would isolate the problem to the upstream kernel update and rule out anything you did.

dangowrt commented:
So this is the cause:
```
[    0.785209] mtk-pcie 1a143000.pcie: deferred probe timeout, ignoring dependency
[    0.792549] mtk-pcie: probe of 1a143000.pcie failed with error -110
[    0.798958] mtk-pcie 1a145000.pcie: deferred probe timeout, ignoring dependency
[    0.806271] mtk-pcie: probe of 1a145000.pcie failed with error -110
[    0.812615] mtk_hsdma 1b007000.dma-controller: deferred probe timeout, ignoring dependency
[    0.820877] mtk_hsdma: probe of 1b007000.dma-controller failed with error -110
[    0.828261] mtk_soc_eth 1b100000.ethernet: deferred probe timeout, ignoring dependency
[    0.836182] mtk_soc_eth: probe of 1b100000.ethernet failed with error -110
[    0.843146] xhci-mtk 1a0c0000.usb: deferred probe timeout, ignoring dependency
...
[    7.540060] mt7622-wmac 18000000.wmac: deferred probe timeout, ignoring dependency
[    7.547699] mt7622-wmac: probe of 18000000.wmac failed with error -110
```
I go hunt that probe dependency issue...

justinkb commented:
Oh damn, that is so neat! does putty support that? otherwise I need to install another tool on my windows laptop that I have the dongle attached to

dangowrt commented:
Found a likely culprit candidate:
```
commit 9de2b9286a6dd16966959b3cb34fc2ddfd39213e
Author: Jiasheng Jiang <jiasheng@iscas.ac.cn>
Date:   Wed Dec 22 09:51:57 2021 +0800

    ASoC: mediatek: Check for error clk pointer
    
    Yes, you are right and now the return code depending on the
    init_clks().
    
    Fixes: 6078c651947a ("soc: mediatek: Refine scpsys to support multiple platform")
    Signed-off-by: Jiasheng Jiang <jiasheng@iscas.ac.cn>
    Link: https://lore.kernel.org/r/20211222015157.1025853-1-jiasheng@iscas.ac.cn
    Signed-off-by: Mark Brown <broonie@kernel.org>
```

justinkb commented:
do you know how that commit should be corrected or should I try the build with it just reverted?

btw, how can I avoid losing my settings when flashing a sysupgrade via the recovery? I just unbricked but I lost my config

dangowrt commented:
First of all, to be sure we are dealing with the right place, the best would be to just apply a revert-commit added to `target/linux/mediatek/patches-5.10`. If that really fixes it, we'll need to add some more debugging code (some kprintf's in mtk-scpsys.c) to figure out *why* probing of the `<&topckgen CLK_TOP_HIF_SEL>` fails at this point.
If it doesn't fix it, then probably there is really something wrong with the clock driver, but that hasn't been modified since 2019...

justinkb commented:
https://lore.kernel.org/all/trinity-2a727d96-0335-4d03-8f30-e22a0e10112d-1643363480085@3c-app-gmx-bap33/

It seems Frank Wunderlich already figured out that is the cause and that reverting it works.

dangowrt commented:
> btw, how can I avoid losing my settings when flashing a sysupgrade via the recovery? I just unbricked but I lost my config

You have to backup your configuration before (e.g. using `sysupgrade -b /tmp/backup.tgz`) as it cannot be kept when flashing from recovery firmware.

dangowrt commented:
> It seems Frank Wunderlich already figured out that is the cause and that reverting it works. (at least for bananapi, but I'm assuming the exact thing is at play here)

Yes, so that will be it then. Now we only need to figure out *why* because just reverting it is not really a good option.

justinkb commented:
is it possible the soc as used in router devices has some clocks that are always disabled? these are the clock names

```
	"mm",
	"mfg",
	"venc",
	"venc_lt",
	"ethif",
	"vdec",
	"hif_sel",
	"jpgdec",
	"audio"
```

if that's the case, the scp_init aborts early on any failed clock producer initialization

justinkb commented:
I'm 99.9% sure this is just a bad commit that should never have been put in any kernel tree.

refer to https://www.mediatek.com/products/products/home-networking/mt7622

there are two versions of the soc:

```
MT7622A: Full feature set
MT7622B: Router-only capabilities
```

this patch breaks stuff for MT7622B

dangowrt commented:
In `mt7622.dtsi` only the `hif_sel` is used for scpsys. `hif` subsystem in mtk SoCs is the high-speed differential PHY for USB3 and PCIe, it should for sure be present in all variants of the chip (I believe the main difference between MT7622A and MT7622B is the package and pin-count, it's maybe even the same die...)

justinkb commented:
I think I figured it out, I've posted my theory on the kernel mailing list under Frank's bug report (hopefully my mail client didn't fuck it up)

This is what I sent:

I guess this breaks all MT7622 SoCs since it'll prematurely exit init_clks (and subsequently init_scp) completely once devm_clk_get fails to get a reference to the mm clock producer (which happens to be the first one tried). This is because MT7623 has a GPU (so no mm clock) and MT7622 doesn't, and as a result the other clock producer pointers never get initialized (and other stuff in init_scp after returning from the error never happens).

The patch seems fundamentally flawed, I guess it was either not tested at all, or only tested on a MT7623. The initialization functions seem designed with the idea that it's ok if some clocks aren't present, so stopping the initialization when one of them isn't present seems wrong. (For example, there is also a MT7622B variant of the MT7622 which probably also lacks some clocks MT7622(A) does have).

justinkb commented:
I juist built master with this PR on top, and the ASoC: mediatek: Check for error clk pointer commit reverted. I can get an ip from the router now, and I can ping it, but it seems the web interface doesn't come up properly. How to best diagnose such a problem? I can post serial logs again

dangowrt commented:
@justinkb That's the expected result if you build from source and don't fetch luci feed before. If you want to build incl. the web-ui, do
```
scripts/feeds update -a
scripts/feeds install -a
make menuconfig
```
and select a LuCI bundle there.

Regarding the reverting of that patch: Good to see that this was really the cause. However, I don't believe we should drop that patch. If the clock is not needed, we can remove it from DTS instead of reverting the patch, see [above](https://github.com/openwrt/openwrt/pull/4975#issuecomment-1024267063). If you can try that, that would be nice.

justinkb commented:
re: your edit, exactly that's why I'm thinking the patch is just bad. The initialization procedure refers to hardcoded clock names (that not all socs using that initialization will all have), so prematurely stopping is just incorrect imo.

justinkb commented:
can confirm I got the webui and everything working on 5.10.94 now. I am gonna try it as my main router over the weekend (I can't afford breakages on weekdays, so nice moment to put it to the test)

graysky2 commented:
Nice work figuring this out @justinkb @dangowrt.  I just patched master + this PR and my RT3200 works as expected.
```
# uname -a
Linux nouveau 5.10.94 #0 SMP Sat Jan 29 00:38:47 2022 aarch64 GNU/Linux
```

So
```
Build system: x86_64
Build-tested: bcm2711/RPi4B, mt7622/RT3200
Run-tested: bcm2711/RPi4B, mt7622/RT3200,
```

graysky2 commented:
> is there any way to to specify before a build to include the exact same package options (incl luci) as the snapshots? Do I really need to figure all that out myself?

@justinkb - I emailed you answering your question to keep the PR conversation on topic.

graysky2 commented:
@rsalvaterra - Seems to be a touchless rebase for 5.10.95, when you finish your run, see mine if you want to compare: https://github.com/graysky2/openwrt/commit/7b0042b75dc959fcf7a6c4f66d9e5f007689c757

rsalvaterra commented:
> @rsalvaterra - Seems to be a touchless rebase for 5.10.95, when you finish your run, see mine if you want to compare: [graysky2@7b0042b](https://github.com/graysky2/openwrt/commit/7b0042b75dc959fcf7a6c4f66d9e5f007689c757)

Thanks, bumping here now, will compare. Also, thanks for the detective work, @dangowrt and @justinkb!

justinkb commented:
5.10.95 running okay here

rsalvaterra committed:
kernel: bump 5.10 to 5.10.95

Patches automatically rebased.

Signed-off-by: Rui Salvaterra <rsalvaterra@gmail.com>

rsalvaterra commented:
Automatic rebase (two patches), maching your result, @graysky2.

graysky2 commented:
5.10.95
```
Build system: x86_64
Build-tested: bcm2711/RPi4B, mt7622/RT3200
Run-tested: bcm2711/RPi4B, mt7622/RT3200
```

rockdrilla commented:
friendly ping to @rsalvaterra: there's 5.10.96 omg %)

rsalvaterra commented:
Thanks for the heads up, @rockdrilla. 😉

graysky2 commented:
Again, seems to be hands free.  Compare to my result: https://github.com/graysky2/openwrt/commit/7c1fca953f0303e130b4412a7c50bbd029cee0b9

rsalvaterra committed:
kernel: bump 5.10 to 5.10.96

Patches automatically rebased.

Signed-off-by: Rui Salvaterra <rsalvaterra@gmail.com>

graysky2 commented:
@dangowrt - I applied the 5.10.95 to 5.10.96 commit from my own rebase (https://github.com/graysky2/openwrt/commit/7c1fca953f0303e130b4412a7c50bbd029cee0b9) and found that the network on the RT3200 would not come up (wired or wireless).  Same symptoms I experienced above.  Any thoughts?

rsalvaterra commented:
@graysky2, your bump matches mine, thanks!

> @dangowrt - I applied the 5.10.95 to 5.10.96 commit from my own rebase ([graysky2@7c1fca9](https://github.com/graysky2/openwrt/commit/7c1fca953f0303e130b4412a7c50bbd029cee0b9)) and found that the network on the RT3200 would not come up (wired or wireless). Same symptoms I experienced above. Any thoughts?

Oh, no. Not again…! 🤦

dangowrt commented:
@rsalvaterra you have dropped the [patch reverting the culprit commit](https://github.com/openwrt/openwrt/commit/93bb4b6b3394f9ae23e87edfce3664b597eb14a4#diff-b19fd183892de0e67ded003b491d50a1b21bd114915628c6bff695b2f083352d) which I had pushed into your tree...

rsalvaterra commented:
I'm an idiot. Sorry about that, guys, should be fixed now. 😅

graysky2 commented:
@dangowrt - I am happy this project providing the revert but has someone communicated with the kernel ML regarding this?  Thought I saw someone posted a link but I did not follow it.

dangowrt commented:
@graysky2 Yes, some noise has been made upstream:
https://lore.kernel.org/all/CAOn50SJrdS+uRnJ=e1vrb6n5e1TSWLBiLDdZe=7tSCXRVwtTjQ@mail.gmail.com/

hauke commented:
Thank you for your patches, @dangowrt  applied them to master in f2232544b879affe4e84c13fdf38da80e07c7360, 1343acc8cd9e5c4206c2f65f1897c7be5660f046, b9736cb78c508dfeda7178973fe4fb9eb12c6a1b and ff97fb98e3176aef0378318d7ff52aaa85bfeb48
